### ISSUE TITLE
get[Date]AsDate functions throw invalid Exception

### ISSUE THREAD

**Issue by IanSimpson** [XeroAPI/xero-php-oauth2#189]
**SDK you're using (please complete the following information):**
 - Version 2.2.0

**Describe the bug**
Since 2.0 all the classes (e.g. Invoice) include functions to get each date as a date (`getDateAsDate`, `getFullyPaidOnDateAsDate`, etc.)

These try to convert the string date to a DateTime object. If the string date is null they try to throw an Exception:

```
throw new Exception('can not convert null string to date');
```

This Exception in turn generates a fatal error:

```
Uncaught Error: Class 'XeroAPI\\XeroPHP\\Models\\Accounting\\Exception` not found
```

Presumably it should either be

```
throw new \Exception('can not convert null string to date');
```

or have a `use \Exception;` at the start of each file

**To Reproduce**
1. Pull an unpaid invoice from the API
2. Run `$invoice->getFullyPaidOnDateAsDate()`
3. Observe fatal error

**Expected behavior**
I'd expect and Exception (which I can catch) to be thrown instead of a fatal error

**Comment by SidneyAllen** [comment]
@IanSimpson - if you have time to make a PR or share an example of how you'd recommend implementing this we can look to add it to the templates we use for generating the SDK.

**Comment by SidneyAllen** [comment]
Closing due to in activity

**Comment by IanSimpson** [comment]
I didn’t see the comments here before sorry. A bit rubbish this is just getting abandoned. I’ve even given you two possible solutions. I’m not quite sure how to implement best in this context so can’t really do a PR.

**Comment by SidneyAllen** [comment]
Issue resolved in version 2.4.3
https://github.com/XeroAPI/xero-php-oauth2/releases/tag/2.4.3
